datafusion/common/src/scalar.rs

```diff
@@ -1385,24 +1385,37 @@ impl ScalarValue {
                 .map(|s| s.to_array())
                 .collect::<Result<Vec<_>>>()?;
 
-            let capacity = Capacities::Array(arrays.iter().map(|arr| arr.len()).sum());
+            let capacity = Capacities::Array(
+                arrays
+                    .iter()
```

Comment:
was thinking if filter_map will be handy here to avoid double iteration? 

```
arrays.iter().filter_map(|arr| {if arr.is_null(0) {None} else Some(arr.len()) })